add tolerations/nodeselector to migration job template (fixes #1774)

ranvit opened the pull request:
##### SUMMARY
As described in https://github.com/ansible/awx-operator/issues/1774, the migration pod has no provisions to add tolerations to it. This PR's solution is to use the postgres pod's toleration/selector on the migration job. 

The docs would need to be updated to alert the user about this relationship between postgres and the migration pod

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

rooftopcellist commented:
@ranvit Is there a reason why `postgres_tolerations` and `postgres_node_selector` is used instead of `task_tolerations` and `task_node_selector`?

My thinking here is that specifying `postgres_` params might not make sense in some scenarios, for example external db scenarios.  

The k8s db migration job uses the task container, and previously this was running in the task container, so using `task_*` params for this might be a bit more intuitive.  

Also, could you add some docs here to make this a bit more discoverable for users?
* https://github.com/ansible/awx-operator/blob/devel/docs/user-guide/advanced-configuration/assigning-awx-pods-to-specific-nodes.md

ranvit committed:
Enable fallback to global settings for db-migration job scheduling (#1804)

Modified the db-migration job template to use `task_*` settings with a fallback to global AWX configurations if not specified.

ranvit commented:
@rooftopcellist thank you for the feedback. It makes sense for this migration job to utilize the task_* settings. 

I decided to implement this the same way the Task pods have been configured -- with a fallback to the global awx settings for node assignment. Also updated the docs!

rooftopcellist reviewed (approved):
I tested this out on my cluster and it works as expected as far as I can tell.  No errors on fresh install either.  

@ranvit Thanks for the PR!

ranvit commented:
@rooftopcellist how do you feel about adding the same tolerations/nodeSelector fallback on the Backup role's [db-management pod](https://github.com/ansible/awx-operator/blob/devel/roles/backup/templates/management-pod.yml.j2)?

I could go through every k8s pod template in the repo and add the same fix
